### PR TITLE
Updating to use SDK 3 & async patterns.

### PR DESCRIPTION
**Pull request by MongoCaleb**
Using the same async pattern we're using in other Tutiorials.

### CODE REVIEW

**Comment by MongoCaleb** [comment]
@adamchel can you take a look and/or assign someone else to? Thanks!

File: StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MyGCMListenerService.java [hunk 1]
```diff
@@ -4,7 +4,9 @@
 import android.app.NotificationManager;
 import android.content.Context;
 import android.os.Bundle;
+import android.support.v4.app.NotificationCompat;
 import android.util.Log;
+import android.widget.Toast;
```
Comment (adamchel) [inline StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MyGCMListenerService.java]: [nit] I don't think you need this.

File: StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java [hunk 2]
```diff
@@ -17,11 +17,13 @@
 import com.google.android.gms.tasks.Task;
 import com.mongodb.stitch.android.StitchClient;
 
+import com.mongodb.stitch.android.auth.AvailableAuthProviders;
```
Comment (adamchel) [inline StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java]: [nit] and actually this one doesn't look necessary either

File: StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java [hunk 3]
```diff
@@ -17,11 +17,13 @@
...
 import com.mongodb.stitch.android.auth.anonymous.AnonymousAuthProvider;
 import com.mongodb.stitch.android.push.AvailablePushProviders;
 import com.mongodb.stitch.android.push.gcm.GCMPushClient;
+import com.mongodb.stitch.android.services.mongodb.MongoClient;
```
Comment (adamchel) [inline StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java]: [nit] ditto

**Pull request merged**